clouddriver-ecs/src/integration/java/com/netflix/spinnaker/clouddriver/ecs/test/CreateServerGroupForExistingServiceSpec.java

```diff
@@ -0,0 +1,323 @@
+import com.amazonaws.services.ecs.model.*;
+import com.amazonaws.services.elasticloadbalancingv2.AmazonElasticLoadBalancing;
+import com.amazonaws.services.elasticloadbalancingv2.model.DescribeTargetGroupsRequest;
+import com.amazonaws.services.elasticloadbalancingv2.model.DescribeTargetGroupsResult;
+import com.amazonaws.services.elasticloadbalancingv2.model.TargetGroup;
+import com.netflix.spinnaker.clouddriver.artifacts.ArtifactCredentialsRepository;
+import com.netflix.spinnaker.clouddriver.artifacts.ArtifactDownloader;
+import com.netflix.spinnaker.clouddriver.artifacts.config.ArtifactCredentials;
+import com.netflix.spinnaker.clouddriver.aws.security.NetflixAmazonCredentials;
+import com.netflix.spinnaker.clouddriver.ecs.EcsSpec;
+import com.netflix.spinnaker.kork.artifacts.model.Artifact;
+import io.restassured.http.ContentType;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.stubbing.Answer;
+import org.springframework.boot.test.mock.mockito.MockBean;
+
+public class CreateServerGroupForExistingServiceSpec extends EcsSpec {
+
+  @MockBean ArtifactDownloader mockArtifactDownloader;
+
+  @MockBean ArtifactCredentialsRepository mockArtifactCredentialsRepository;
+
+  private ArtifactCredentials mockArtifactCredentials = mock(ArtifactCredentials.class);
+
+  private AWSApplicationAutoScalingClient mockAWSApplicationAutoScalingClient =
+      mock(AWSApplicationAutoScalingClient.class);
+
+  private AmazonECS mockECS = mock(AmazonECS.class);
+
+  private AmazonElasticLoadBalancing mockELB = mock(AmazonElasticLoadBalancing.class);
+
+  @BeforeEach
+  public void setup() {
+
+    // mocking calls
+    when(mockECS.describeServices(any(DescribeServicesRequest.class)))
+        .thenReturn(
+            new DescribeServicesResult()
+                .withServices(
+                    Collections.singletonList(
+                        new Service()
+                            .withServiceName("createService")
+                            .withCreatedAt(new Date())
+                            .withStatus("INACTIVE"))));
+
+    when(mockECS.createService(any(CreateServiceRequest.class)))
+        .thenReturn(
+            new CreateServiceResult().withService(new Service().withServiceName("createdService")));
+
+    when(mockECS.registerTaskDefinition(any(RegisterTaskDefinitionRequest.class)))
+        .thenAnswer(
+            (Answer<RegisterTaskDefinitionResult>)
+                invocation -> {
+                  RegisterTaskDefinitionRequest request =
+                      (RegisterTaskDefinitionRequest) invocation.getArguments()[0];
+                  String testArn = "arn:aws:ecs:::task-definition/" + request.getFamily() + ":1";
```

Comment:
I don't think this will have downstream effects on other parts of this operation, but if a task definition with this server group name (what would be provided as the "family" param) already exists, then the newly registered and returned task def should have an incremented revision ("2" in this case). Let's update that so it's in line with what is expected in this situation.